R/get_types_recipes.R

```diff
@@ -0,0 +1,112 @@
+## Buckets variables into discrete, mutally exclusive types
+get_types <- function(x) {
+  res <- lapply(x, get_types_recipes)
+  tibble(variable = names(res), type = unname(res))
+}
+
+#' Get types for use in recipes
+#'
+#' The `get_types_recipes()` generic is used internally to supply types to
+#' columns used in recipes. These functions underlie the work that the user sees
+#' in [selections].
+#'
+#' This function acts a extended recipes-specific version of [class()]. By
+#' ignoring differences in similar types ("double" and "numeric") and allowing
+#' each element to have multiple types ("factor" returns "factor", "unordered",
+#' and "nominal". And "character" returns "string", "unordered", and
+#' "nominal") we are able to create more natural selectors such as
+#' [all_nominal()], [all_string()] and [all_integer()].
+#'
+#' @export
+#' @param x A character vector
```

Comment:
`x` can be any type of atomic vector, right? Is this supposed to be the `@return` entry?

Reply:
No this is just plain wrong. fixed in https://github.com/tidymodels/recipes/pull/993/commits/4b91c539c156393942bb2459b13beaae674e3ed7